src/Credentials/EcsCredentialProvider.php

```diff
@@ -84,6 +103,10 @@ public function __invoke()
      */
     private function getEcsUri()
     {
+        $credFullUri = getenv(self::ENV_FULL_URI);
```

Comment:
Dug into it a bit further and it looks like this will need to be refactored.  Other SDKs evaluate the relative uri before the full uri and that's what we'll need to do for sake of consistency.  The [Java docs](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/EC2ContainerCredentialsProviderWrapper.html) give a good outline of the desired logic here.  More than willing to chip in on this if you need any help.

Reply:
done